integrations/google_genai/src/haystack_integrations/components/embedders/google_genai/text_embedder.py

```diff
@@ -0,0 +1,136 @@
+        model: str = "text-embedding-004",
+        config: Optional[types.EmbedContentConfig] = types.EmbedContentConfig(
+            task_type="SEMANTIC_SIMILARITY"),
+
+    ):
+        """
+        Creates an GoogleAITextEmbedder component.
+
+        :param api_key:
+            The Google API key.
+            You can set it with an environment variable `GOOGLE_API_KEY`, or pass with this parameter
+            during initialization.
+        :param model:
+            The name of the model to use for calculating embeddings.
+            The default model is `text-embedding-004`.
+        :param config:
+            A dictionary of keyword arguments to configure embedding content configuration `types.EmbedContentConfig`.
+            For more information, see the [Google AI Task types](https://ai.google.dev/gemini-api/docs/embeddings#task-types).
+        """
+
+        self._api_key = api_key
+        self._model_name = model
+        self._config = config
+        self._client = genai.Client(api_key=api_key.resolve_value())
+
+    def _get_telemetry_data(self) -> Dict[str, Any]:
+        """
+        Data that is sent to Posthog for usage analytics.
+        """
+        return {"model": self.model}
+
+    def to_dict(self) -> Dict[str, Any]:
+        """
+        Serializes the component to a dictionary.
+
+        :returns:
+            Dictionary with serialized data.
+        """
+        return default_to_dict(
+            self,
+            model=self._model_name,
+            api_key=self._api_key.to_dict(),
+            config=self._config.to_json_dict()
+        )
+
+    @classmethod
+    def from_dict(cls, data: Dict[str, Any]) -> "GoogleAITextEmbedder":
+        """
+        Deserializes the component from a dictionary.
+
+        :param data:
+            Dictionary to deserialize from.
+        :returns:
+            Deserialized component.
+        """
+        deserialize_secrets_inplace(data["init_parameters"], keys=["api_key"])
+        return default_from_dict(cls, data)
+
+    def _prepare_input(self, text: str) -> Dict[str, Any]:
+        if not isinstance(text, str):
+            error_message_text = (
+                "GoogleAITextEmbedder expects a string as an input. "
+                "In case you want to embed a list of Documents, please use the GoogleAIDocumentEmbedder."
+            )
+
+            raise TypeError(error_message_text)
+
+        text_to_embed = text
+
+        kwargs: Dict[str, Any] = {
+            "model": self._model_name, "contents": text_to_embed}
+        if self._config:
+            kwargs["config"] = self._config
+
+        return kwargs
+
+    def _prepare_output(self, result: types.EmbedContentResponse) -> Dict[str, Any]:
+        return {"embedding": result.embeddings[0].values, "meta": {"model": self._model_name}}
+
+    @component.output_types(embedding=List[float], meta=Dict[str, Any])
+    def run(self, text: str):
```

Comment:
Let's add a return type here 
```suggestion
    def run(self, text: str) -> Union[Dict[str, List[float]], Dict[str, Any]]:
```